### ISSUE TITLE
Unresponsive terminal pane, using zsh and p10k theme

### ISSUE THREAD

**Issue by jdhoffa** [rstudio/rstudio#7584]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.706 (**This issue does not seem to affect the current release, only the daily build!)
    OS Version      : macOS
    R Version       : 4.0.2

### Steps to reproduce the problem
In terminal:
* Install and configure `powerline10k` zsh theme: https://github.com/romkatv/powerlevel10k

In RStudio: 
* Global Options > Terminal > General > New terminals open with "Zsh"
* Global Options > Terminal > General > Connection> Uncheck Connect with WebSockets

### Describe the problem in detail

I can use RStudio, along with the terminal, for a while but eventually, it becomes unresponsive. The terminal cursor still blinks, but it doesn't respond to keyboard input. 

I can start a new terminal window, and the new window will work, but the old window remains unresponsive.

I have not been able to determine a specific action that causes it to become unresponsive. 

Global Terminal Information: 
```
Global Terminal Information
---------------------------
Loaded TerminalSessions: 1
Handle: 'BD7200B3' Caption: 'Terminal 1'

Terminal List Count: 1
Handle: 'BD7200B3' Caption: 'Terminal 1' Session Created: true

Current Terminal Session Information
------------------------------------
Caption:     'Terminal 1'
Title:       'jdhoffa@jacksons-MBP: ~/Documents/github/2DegreesInvesting/r2dii.analysis'
Cols x Rows  '117 x 26'
Shell:       'Zsh'
Handle:      'BD7200B3'
Sequence:    '1'
Restarted:   'false
Busy:        'false'
Exit Code:   'null'
Full screen: 'client=false/server=false'
Zombie:      'false'
Track Env    'false'
Local-echo:  'true'
Working Dir: '~/Documents/github/2DegreesInvesting/r2dii.analysis'
Interactive: 'Always'
WebSockets:  'false'

System Information
------------------
Desktop:    'true'
Remote:     'false'
Platform:   'Mac'

Connection Information
----------------------
2020/8/14 12:04:24: Connected with RPC

Local-echo Match Failures
-------------------------
<none>
```

Also, I have yet to experience similar behaviour when running the same terminal setup/ theme directly in iTerm, leading me to believe that the problem arises specifically in RStudio. 

### Describe the behavior you expected
I expect the terminal to continue to respond ;-)

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists. 
-> this issue seems to only affect the daily build!
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@jdhoffa Thank you for raising the issue! I'm not able to reproduce it yet, but I'll stick to `zsh` as my default terminal shell to see if I run into the issue.

**Comment by gtritchie** [comment]
I have reproduced this behavior, but not related to the shell or shell-theme. It happens on all platforms, desktop and server. Rather than open a separate issue for what is almost certainly the same thing, here are my details:

    RStudio Edition : Desktop or Server
    RStudio Version : 1.4.542+
    OS Version      : any
    R Version       : any

Note, believe was introduced via #7275 

### Steps to reproduce the problem
- start IDE in normal configuration with Console/Terminal/Jobs tabs showing in Console panel
- click on Terminal, confirm a terminal starts and is responsive (e.g. hit Enter and see a new prompt)
- close the Jobs tab via the little [x]
- switch back to the Terminal and try to use it

### Describe the problem in detail
- terminal is unresponsive; cursor is blinking but nothing happens

### Describe the behavior you expected
- terminal continues to work

If you restart after these steps, then the terminal will work again. If you then redisplay the Jobs pane via View / Show Jobs, the problem comes back. It's not specific to the Jobs pane, if you do View / Show Find in Files it will happen, too.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.4.734, R Version - 4.0.2, Windows 10 x64 - Desktop 

Comparison tested between 1.4.706 and the latest daily. The issue is fixed (Terminal pane no longer becomes unresponsive after adding or removing tabs from the Console panel) and the fix seems stable (tested around this area, multiple terminals, different terminal types, adding removing Jobs and Find in Files tab, etc. and the terminal and other tabs are remaining responsive).
